[Java][Native] Fix multipart builder files array (#16055)

lucar94 opened the pull request:
[Java][Native] Fix multipart builder when API accepts files array. Fix #16055 

<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

wing328 commented:
Thanks for the PR.

Can you please add a test by adding the following endpoint in `modules/openapi-generator/src/test/resources/3_0/echo_api.yaml` ?

```yaml
  # Array of binary in multipart mime tests
  /body/application/octetstream/binary:
    post:
      tags:
        - body
      summary: Test array of binary in multipart mime
      description: Test array of binary in multipart mime
      operationId: test/body/multipart/formdata/array_of_binary
      requestBody:
        content:
          multipart/form-data:
            schema:
              required:
                - files
              type: object
              properties:
                files:
                  type: array
                  items:
                    type: string
                    format: binary
      responses:
        '200':
          description: Successful operation
          content:
            text/plain:
              schema:
                type: string
```

and add a test in `samples/client/echo_api/java/native/src/test/java/org/openapitools/client/CustomTest.java` ?

lucar94 commented:
Hi, i need some help (it's my first contributing on this project).

To `modules/openapi-generator/src/test/resources/3_0/echo_api.yaml` I added the endpoint indicated (path `/body/application/octetstream/binary` already exists, so I modified to `/body/application/octetstream/array_of_binary`).

Then I've run the command `./bin/generate-samples.sh ./bin/configs/java-native*` that has regenerated BodyApi.java (and some other files).

I see that `samples/client/echo_api/java/native/src/test/java/org/openapitools/client/CustomTest.java` is not editable manually. How can I do to generate a new test?

Thanks in advance.

wing328 commented:
> so I modified to /body/application/octetstream/array_of_binary).

yes please 

> I see that samples/client/echo_api/java/native/src/test/java/org/openapitools/client/CustomTest.java is not editable manually.

You can edit this file with a new test and use EchoServerResponseParser to parse the result (response from the server). To run the echo server locally, please refer to https://github.com/OpenAPITools/openapi-generator/wiki/Integration-Tests#echo-server

lucar94 commented:
Thanks, I added the test

wing328 commented:
can you please run `./bin/utils/ensure-up-to-date` to update all samples? python echo client also using that test spec.

thanks.

wing328 commented:
Samples updated via 66155d622cb
